### PR TITLE
Ensure "Device" is not considered as a resource type for R5

### PR DESCRIPTION
**Pull request by lukedegruchy**
- Check for the FHIR version and only add "Device" to the list of patient compartment resources if the FHIR version is within the whitelist.

Closes https://github.com/hapifhir/hapi-fhir/issues/5913

### CODE REVIEW

File: hapi-fhir-jpaserver-test-r4/src/test/java/ca/uhn/fhir/jpa/bulk/BulkDataExportProviderR4Test.java [hunk 1]
```diff
@@ -874,6 +873,7 @@ public void testInitiatePatientExportRequest() throws IOException {
 		assertThat(bp.getResourceTypes(), containsInAnyOrder("Immunization", "Observation"));
```
Comment (jmarchionatto) [inline hapi-fhir-jpaserver-test-r4/src/test/java/ca/uhn/fhir/jpa/bulk/BulkDataExportProviderR4Test.java:873]: Assertion is now redundant

**Review by jmarchionatto — APPROVED** [review]
Looks good! Approved with a minor comment

File: hapi-fhir-jpaserver-test-r4/src/test/java/ca/uhn/fhir/jpa/bulk/BulkDataExportProviderR4Test.java (hunk 1 above)
Comment (lukedegruchy) [reply hapi-fhir-jpaserver-test-r4/src/test/java/ca/uhn/fhir/jpa/bulk/BulkDataExportProviderR4Test.java]: Fixed

**Review by tadgh — APPROVED** [review]

**Comment by tadgh** [comment]
Note I only blind-approved this due to codeowners bug. I did not actually review. Juan did

**Pull request merged**